engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala

```diff
@@ -0,0 +1,137 @@
+import cromwell.engine.EngineWorkflowDescriptor
+import cromwell.engine.workflow.lifecycle.execution.JobPreparationActor._
+import wdl4s._
+import wdl4s.expression.WdlStandardLibraryFunctions
+import wdl4s.util.TryUtil
+import wdl4s.values.WdlValue
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+object JobPreparationActor {
+  /** Commands */
+  case object Start
+  /** Responses */
+  case class BackendJobPreparationSucceeded(jobDescriptor: BackendJobDescriptor, props: Props)
+  case class BackendJobPreparationFailed(jobKey: JobKey, throwable: Throwable)
+  /** States */
+  sealed trait JobPreparationActorState
+  case object Idle extends JobPreparationActorState
+  case object EvaluatingInputs extends JobPreparationActorState
+  case object BuildingProps extends JobPreparationActorState
+  /** Internal Messages*/
+  private case class EvaluatedInputs(inputs: Map[LocallyQualifiedName, WdlValue])
+  private case class BackendActorProps(jobDescriptor: BackendJobDescriptor, props: Props)
+
+  def props(executionData: WorkflowExecutionActorData,
+            jobKey: BackendJobDescriptorKey,
+            factory: BackendLifecycleActorFactory,
+            configDescriptor: BackendConfigurationDescriptor) = Props(new JobPreparationActor(executionData, jobKey, factory, configDescriptor))
+}
+
+case class JobPreparationActor(executionData: WorkflowExecutionActorData,
+                               jobKey: BackendJobDescriptorKey,
+                               factory: BackendLifecycleActorFactory,
+                               configDescriptor: BackendConfigurationDescriptor) extends LoggingFSM[JobPreparationActorState, Unit] with WdlLookup {
+
+  override val workflowDescriptor: EngineWorkflowDescriptor = executionData.workflowDescriptor
+  override val executionStore: ExecutionStore = executionData.executionStore
+  override val outputStore: OutputStore = executionData.outputStore
+  override val expressionLanguageFunctions = executionData.expressionLanguageFunctions
+
+  startWith(Idle, Unit)
+
+  when(Idle) {
+    case Event(Start, _) =>
+      resolveAndEvaluate(jobKey, expressionLanguageFunctions) onComplete {
+        case Success(inputs) => self ! EvaluatedInputs(inputs)
+        case Failure(f) =>
+          context.parent ! BackendJobPreparationFailed(jobKey, f)
+          context stop self
+      }
+      goto(EvaluatingInputs)
+  }
+
+  when(EvaluatingInputs) {
+    case Event(EvaluatedInputs(inputs), _) =>
+      val jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, inputs)
+      val config = BackendConfigurationDescriptor(configDescriptor.backendConfig, configDescriptor.globalConfig)
+
+      Future(factory.jobExecutionActorProps(jobDescriptor, config)) onComplete {
+        case Success(props) =>
+          context.parent ! BackendJobPreparationSucceeded(jobDescriptor, props)
+          context stop self
```

Comment:
I can't remember... but is there some issue with using `context ` within a `onComplete` handler?  @mcovarr?